src/core/unit-varlink.c

```diff
@@ -311,18 +314,114 @@ static int list_unit_one(sd_varlink *link, Unit *unit, bool more) {
         return sd_varlink_reply(link, v);
 }
 
+static int lookup_unit_by_name(sd_varlink *link, Manager *manager, const char *name, Unit **ret_unit) {
+        assert(link);
+        assert(manager);
+        assert(ret_unit);
+
+        if (!name)
+                return -EINVAL;
```

Comment:
This can be an assert, the function should never even be called if we don't have a name

Reply:
I'd argue: NULL string a semantically correct value here. Plus, I made is symmetric with machined's varlink.
I suggest to keep as is unless strong objections